src/app/shared/directives/tooltip.directive.ts

```diff
@@ -0,0 +1,63 @@
+import { ApplicationRef, ComponentRef, createComponent, Directive, EventEmitter, HostListener, Input, OnDestroy, Output, TemplateRef } from '@angular/core'
+import { TooltipComponent } from '../components/tooltip/tooltip.component'
+
+export interface TooltipMetadata {
+  xPosition: number
+  width?: number
```

Comment:
I am a bit confused regarding this property, and where it is set and what it is the width of.

Reply:
It is the width of the element that the tooltip is a tooltip for. So I guess you could call it `hostElementWidth`?

The reason it's on this interface is because the Hoverscrub needs to know the width of the element displaying the video in order to convert the mouse position into the specific timestamp in the video. And the element displaying the video is not necessarily available where we create the markup for the tooltip (this is the case for PieceTooltipComponent).

I would say the width of the host element for the tooltip make sense to have on TooltipMetadata.

I have updated it to be called `hostElementWidth`

Reply:
I don't think that it should be a part of the `tooltipMetadata`, since hoverscrub has nothing to do with the tooltip directive or the tooltip component. It is content that is inserted. What the tooltip directive outputs is the option to listen to changes in the position within the element.

The width is set in the same components as the value is extracted again. Instead of having MiniShelfComponent and OffsetablePieceComponent adding to the tooltipMetadata object, they could compute the hoverscrub percent to use directly and then discard the tooltip metadata.

Reply:
It has been removed